substrate/frame/revive/src/lib.rs

```diff
@@ -547,13 +620,81 @@ pub mod pallet {
 	}
 
 	#[pallet::hooks]
-	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T>
+	where
+		// We need this to place the substrate block
+		// hash into the logs of the receipts.
+		T::Hash: frame_support::traits::IsType<H256>,
+		// We need these to access the ETH block gas limit via `Self::evm_block_gas_limit()`.
+		<T as frame_system::Config>::RuntimeCall:
+			Dispatchable<Info = frame_support::dispatch::DispatchInfo>,
+		BalanceOf<T>: Into<U256> + TryFrom<U256>,
+		MomentOf<T>: Into<U256>,
+	{
 		fn on_idle(_block: BlockNumberFor<T>, limit: Weight) -> Weight {
 			let mut meter = WeightMeter::with_limit(limit);
 			ContractInfo::<T>::process_deletion_queue_batch(&mut meter);
 			meter.consumed()
 		}
 
+		fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
+			ReceiptInfoData::<T>::kill();
+			EthereumBlock::<T>::kill();
+			InflightEvents::<T>::kill();
+			InflightTransactions::<T>::kill();
+
+			Weight::zero()
+		}
+
+		fn on_finalize(block_number: BlockNumberFor<T>) {
+			// If we cannot fetch the block author there's nothing we can do.
+			// Finding the block author traverses the digest logs.
+			let Some(block_author) = Self::block_author() else {
+				// Drain storage in case of errors.
+				InflightEvents::<T>::kill();
+				InflightTransactions::<T>::kill();
+				return;
+			};
+
+			// Note: This read should be accounted for.
```

Comment:
no need for these comments here, you can mention that the weight of this function is accounted for on on_intitialize